New Story: Philosopher's Paradise (Replaces "Hai Slaves")

NomadicVolcano opened the pull request:
**Edit: I'm going to merge a bunch of changes MasterOfGrey and I have been working on from the Hai Reveal work over to this branch over the next few days. I'm temporarily changing this back to a Draft PR to make sure it isn't reviewed until then.**

Researchers have been slowly vanishing over the past few decades. When Captain `<last>` discovers where they've gone, it brings the Hai and humans to the brink of war. Only through a daring rescue with the help of two governments can peace be assured.

The commits also add the beginning of the Small Magellanic Cloud expansion: the Whispering Void, a forest just north of Coalition space that starts the intergalactic expanse between the two galaxies. Credit goes to Righthand for placing all systems whose names begin with, "Deep Space."  (Edit: I added two "Deep Space" systems west of Devil-Hide so patrol ships have somewhere to go instead of immediately landing.)

This replaces PR #4900, the Hai Slaves story. That story was based on the premise that pirates were enslaving Hai. That's such a phenomenally stupid thing for pirates to do that it reduces the story's premise to, "pirates are stupid," which doesn't make for a very good story. Try as I might, I couldn't make it interesting without major plot holes, so I changed the premise instead.

Relevant save file:

- [Omega Turtle~6000 BEGIN.txt](https://github.com/endless-sky/endless-sky/files/4371092/Omega.Turtle.6000.BEGIN.txt)

There are five planets with landing missions to rescue a Hai from human space. Do at least three of the five missions: Moonshake, Shangri-La, Stormhold, Canyon, and Skillet. About thirty days after your next landing, you'll be told to go to Darkwaste.

You may see the shipyard and Nanachi payment missions if you take long enough. Other Hai missions are complete.

NOTE: Out of necessity (to set up lore) PR #4947 is merged into this PR. If this PR is accepted in its entirety, then that one has been as well.

## PR Checklist
 - [X] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [X] I uploaded the necessary image, blend, and texture assets here: https://drive.google.com/open?id=1kPTqRlqCAqIqnW24j6hizfn2NV8iqC0k
 - [X] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/120

Zitchas commented:
Sounds like a neat concept. I look forward to reading it. (and the pincer ship looks cool, too)

NomadicVolcano commented:
> Sounds like a neat concept. I look forward to reading it. (and the pincer ship looks cool, too)

The pincer is the first of several critters that will be in that region of space. It is a forest outside the Drak sphere of influence, completely untouched by civilization until a century ago.  The creature has a descriptive name because there is no intelligent species around to have given it a "proper" name previously.

Zitchas commented:
I like more space-faring creatures of the natively-fly through space variety.

NomadicVolcano commented:
I plan on making an ecosystem with plants, prey, predators, and apex predators. That's my next project after the Korath Exiles expansion.

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
~~talking mad shit for someone who coulda been blown out of the sky by first last in several missions~~

I get what she's trying to do, but to me it's unnecessary

NomadicVolcano replied:
You need to tell me which line you're referring to; I don't see this on line 1202 and I am unable to deduce the location.

Arachi-Lover replied:
line 1518

NomadicVolcano replied:
Actually, her treatment of you make no sense given how the General talks to you later. She should speak to you with respect since she knows she needs the Unfettered to respect you. I'll find a way to change the dialog. Thanks' for catching this.

Arachi-Lover replied:
I got the impression that she was trying to say "see? I'm just like you, trust me, follow my lead, do what I say, I'm good at Unfettering" to the Unfettered with the action. good to know it was an issue though, yeah did find it somewhat abrupt how the guydude later was buddybuddy with you while she did that out of nowhere.

NomadicVolcano replied:
I revised it, and some later text that was related. Please let me know if the new version is better.

Arachi-Lover replied:
yeah, it's much better now.

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
yeah so I don't think my laptop can run that very comfortably. maybe tone down the fleet for folks without very good PCs

NomadicVolcano replied:
I will have to ponder this. The fleet needs to look impressively large. I suspect the problem isn't the number of ships, but the number of active objects on the screen.  These ships fire a lot of missiles every time a pirate shows up. Some ships have fighters too, which magnifies the problem. Perhaps Cruisers, Protectors, and Dreadnoughts with special outfitting would solve the problem.

Arachi-Lover replied:
yeah the main issue are the Navy drones since there'd be a lot of them. The sum of all ships there isn't quite the issue on it's own aside from that, but considering that's just the allied fleet, and there'll be the whole fight with a ton of Hostiles once that part of the whole thing is reached, I could see frames on stuff like my own dropping by a lot. Though, I'm afraid don't have many suggestions on how to help with that

NomadicVolcano replied:
My thinking on this matter is to use a fleet of ships purpose-built to kill Hai-equipped Shield Beetles. I've never tried to do that though, so I will have to think about it a while longer.

NomadicVolcano replied:
I've done the best I can to reduce this problem in my latest commit. I tested this by running lots of big things in the background on an old desktop machine.

1. No grasshopper squadrons and few lightning bugs at Waypoint
2. Combined human fleets are in a temporary "Human Alliance" government that uses no projectiles, fighters, or drones.
3. Reduced from 60 fleets to 30 for armada that escorts the upgraded unfettered ships

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
the Hai ships below (and apparently the mission before) number in 14 Beetles, though the log entry and dialog say 15, am I missing something obvious?

NomadicVolcano replied:
The Hai have one ship outfitted with Korath and Wanderer outfits. The other fourteen have Hai weapons which are later replaced by human weapons. Hence, fifteen in total.

Arachi-Lover replied:
ah. so I missed something obvious. I see :james:

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
`hanger` -> `hangar`

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
`With the current anti-human sentiment in both sides of Hai space, Hai unification (peaceful or not) may end poorly for us.` can be removed. I've read this and frankly haven't gotten the impression of a strong anti-human sentiment, and even if it were written here, I'd disagree with it, as the Hai aren't robots that would turn to hate all humans due to the misdeeds of some. they even welcome back Unfettered into their territory prior to this mission chain existing, so I don't see a radical xenophobic feeling toward humanity brewing. that, and I doubt the Quarg would not allow persecution of a group in Hai space just like that.

NomadicVolcano replied:
This sentence was unnecessary to the story, so I removed it.

Arachi-Lover commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
lol

NomadicVolcano replied:
You need to tell me which line you're referring to; I don't see this on line 1202 and I am unable to deduce the location.

Arachi-Lover replied:
1811. not exactly an issue, I just felt the need to lol

NomadicVolcano replied:
Right, that. I couldn't resist putting that in.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+		conversation
+			`As you set your landing vector, you detect a distress call directed at your ship. Your scanners show it comes from a remote island. Do you want to check it out?`
+			choice
+				`Yes.`
```
I rewrote all of the "yes" and "no" lines to have something more relevant.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+			`	"Excellent! Everyone pile onboard. Let's go back to <destination>!"`
+				accept
+	on complete
+		log `Found some gullible Hai that pirates tricked into a, "prison role-play cruise," complete with security guards and a brig. They overpowered the pirate crew who crashed the ship and used escape pods. I returned the Hai to <destination>.`
```
I think I fixed all of these.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+		log "People" "Teeneep Goot'r" `Teeneep is a civil rights leader among the Hai who heads an organization with hundreds of employees and many sponsors. She's used her extensive resources to search for answers about kidnapped scientists.`
+		log `The Hai you rescued are among many that have gone missing over the past few decades, along with humans as well. They all have nanotransceivers that are not of human or Hai origin. Teeneep, a Hai civil rights leader, used her network of information-gatherers to find contacts to help you research the problem in human space and beyond. Your search begins with the Quarg south of Hai space.`
+		conversation
+			`You follow the landing coordinates to a flat area of hardened dirt near a cave. A Hai Centipede is parked next to you; you notice it has non-standard outfits. Its shields and weapons are both upgraded, and you can see some human outfits, including a Scram Drive, extra fuel, and a Catalytic Ramscoop. The ship is outfitted to fly rapidly through unfriendly territory without stopping.`
```
I rewrote those sentences so they make more sense.

NomadicVolcano commented:
**WARNING: SPOILERS**

There will be questions about the kidnapping rate and the Hai population of Philosopher's Paradise (20,000 Hai).

**120 years of exponential growth from 60.** Recall that Philosopher's Paradise was founded 120 years ago by denizens of Freedom after capturing a Hai vessel and some Hai. They sent many ships against two Hai vessels and lost most of their fleet. They captured an advanced ship and about 60 Hai. The (now extremely inbred) descendant of those Hai are on Philosopher's Paradise. The philosophers say their population has a prodigious birth rate, so if we assume it triples every twenty years then 60 becomes 43,740. Hence, 20,000 people without further kidnapping is not a problem. Rodents birth in litters, so six children per woman every 20 years is a very low estimate.

**Constant kidnappings** Ever since humans had regular access to Hai space, the pirates who run Philosopher's Paradise have been kidnapping Hai as quickly as they can. It isn't a sudden crime spree that would be easy to detect, and there are numerous individuals doing it (not one group). I'm unable to find solid information on when the humans gained access. 

I'm assuming about 10,000 of Hai were kidnapped over the time humans have been welcome in Hai space. (10 years? 30?) Clearly though, that isn't necessary for the 20,000 population to be right.

**One in a million per year** Most people underestimate how frequently people go missing in the real world. In the US alone, it is about 1 in 10,000 people per year. Most turn up soon after, usually alive, sometimes dead. Some don't turn up. Let's assume it happens a hundred times less often among the Hai because... utopia? That means one out of a million each year.

**32 billion Fettered Hai** The Hai homeworld has 8 billion people, they have many other planets, and they'e been in space at least 90,000 years. Let's pessimistically assume only 3/4 of them moved off world in 90 millennia because... nesting instinct? That means there are at least 32 billion.

**1000 successes per year** By this math, 32,000 people go missing every year. If, say, 1,600 people a year are researchers (5%) and 600 escape, 1000 are captured per year. 

**150 successful crimes per year** That's 2.73 people per day. Most of the victims you find were in groups (13 out of 15) so it's probably about 3 successful crimes per week. One hundred fifty a year.

To entirely prevent these kidnappings, the Hai would need to prevent those 150 crimes per year out of their 32 billion people. Even a utopian surveillance state isn't that good. That's with the most pessimistic math possible. 

If you use more reasonable assumptions:

- 1 in 100,000 kidnapped per year (10x less often than US)
- 200 billion Hai
- 5% are researchers
- 90% success rate by professional kidnappers

You get 90,000 per year.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+			label anyone
+			`	A Hai near Teeneep responds. "Yes... and no. Apart from our own people, we're regularly in contact with police. They investigate the disappearances, but eventually they have to give up and work on other crimes.`
+			`	"Our biggest problem is the lack of help in human space. Local police are generally overwhelmed with crime in the regions where the kidnapping victims are found. There, kidnappings are commonplace. In the areas where the victims were taken, there's virtually no evidence. Your navies and intelligence agencies haven't the time to help us; they're busy cleaning up after the war. There is a lot  ...
+			`	"Since this problem is tiny compared to the total crime in Hai space, the Hai Elders haven't made it a large priority. They tell me that, if we can prove the kidnappings are from a coordinated effort, they'll put significant resources into solving the problem. So far, they say, their investigators have been unable to prove that.`
```
I respond to this in detail below. I also respond to the inevitable question of the Hai population of Philosopher's Paradise.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+			`	You recall Teeneep gave you a contact at Tarazed Shipyards.`
+				accept
+
+ship "Corvette" "Corvette (Hai Trafficker)"
```
I made that change for all but the two ships that are only useful for this story.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+		"Corvette (Speedy)"
+		"Corvette (Missile)"
+
+fleet "Free Worlds Wormhole Guards"
```
The purpose is to show it is a unified effort across all of human space. I made Free Worlds and Deep Security both less frequent.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	to offer
+		has "Hai Rescue [B01] Election: done"
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
```
My choice of words is intended to spark that debate. Would the humans be overrun? That is why I use the word, "might."

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
The fleet movement began when you were away visiting the philosophers several missions ago. I updated the conversation with the Marshal and Agent to tell the player that.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
You're placing all of your comments on line 1202 now, so I can't be certain of what you're talking about. There are two missions with "on enter" for Ultima Thule and Waypoint but you only ever do one of those two missions.

Arachi-Lover replied:
I see, I must have missed a part in the conditions

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
I have changed from the last story so it is now the stated opinion of one man, rather than the game telling the player that these are facts. The player can form their own opinion based on his arguments.

Arachi-Lover replied:
fair enough

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
I fixed most of those, but unless I'm gravely mistaken, you don't capitalize in this situation:

- "My feet are green," he said, "because I never wash them."

The "b" in "because" is not capitalized since the sentence has not yet ended.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
You're placing all of your comments on line 1202 now, so I can't be certain of what you're talking about. I looked through the entire file and corrected all the instances I could find where there were double-quotes incorrectly placed.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
Fixed.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
That should have said, "will discover the secret of Jump Drives for them." The Quarg is being intentionally vague as to how that would happen, to avoid giving the Hai and humans any ideas. Once the Unfettered Hai have all Wanderer, Hai, and human systems, *and* an unlimited supply of Jump Drives, the rest of the galaxy is finished. The Drak decide to intervene to prevent this, using a ball and a worm.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
I removed the time reference and changed it so Captain `<last>` refers to how the Quarg haven't figured it out yet.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
Yes. The captured Korath explain later how they were captured during a raid. (You just haven't gotten to that part yet.)

Arachi-Lover replied:
ya read it later down, forgot to delete this comment.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	on enter "Waypoint"
+		dialog `There is a massive fleet of Hai ships here, and more coming in all the time. If this fleet were to enter human space, we might be overrun.`
+	on enter "Ultima Thule"
+		dialog `Although it is reassuring to see humans come together for a common cause, you don't look forward to seeing two species enter an all-out war over a misunderstanding. You also wonder how the human authorities are keeping the Hai a secret with 
```
You need to tell me which line you're referring to; I don't see this on line 1202 and I am unable to deduce the location.

Arachi-Lover replied:
line 1486, `The Hai Elders are surprised and disturbed when they hear your account of what happened. They give you plans for Shield Beetle upgrades that contain a mix of human and Hai outfits. They also give you a purchase order with an account number for the outfitters to charge.`

NomadicVolcano replied:
You were told to talk to the Hai Elders to get designs for the Shield Beetles. Either the Unfettered already sent the specifications, or the solution is obvious to the Hai. Regardless, the Unfettered wanted the Hai Elders to foot the bill and provide that account number for the purchases.

I assume Teeneep's request to talk to the Hai Elders was also a way to get a detailed report back to them without having to go through an untrusted intermediary. I prefer to leave that point ambiguous though.

Arachi-Lover replied:
nvm, somehow I mistook that `on stopover` for an `on offer`

Arachi-Lover commented:
fair point on them breeding up to a high population.

still not sold on the high kidnappings though, mainly because I don't see it being treated as or regarded as a minor issue in comparison to the other (most likely very rare ocurrences of) crime in Hai space, so I'd still prefer if the mentioned number was reduced. Even if today many go missing, in the ES space age and especially in Hai territory I'd expect them to have a very good intel network for at least knowing if/when/how/why someone went missing in the vast majority of cases.

NomadicVolcano commented on data/smc/missing researchers.txt:
```diff
@@ -0,0 +1,2174 @@
+	to offer
+		has "Hai Rescue [A02] Nanotransceiver Identification: done"
+	on offer
+		log `The Korath identify the nanotransceivers as their race's design, but extremely primitive and poorly-made. The Quarg provided a data card that Tarazed Shipyards may be able to use.`
```
Fixed.

NomadicVolcano commented:
I haven't fixed the problem of the slowness of the missions with the vast fleets. I'll ponder that and fix it this evening.

Arachi-Lover commented:
>There is no lore in-game specifically showing the frequency of crime in Hai space

maybe not exact numbers or data, but the human you encounter that tells you about the Hai says `most of us came here to escape from the chaos and fighting in human space.`  and that `Hai society is so advanced, they need almost no resources to maintain it.` , so I'd say it's safe to say the crime there is much rarer than Human space.

>Even with those assumptions, 1000 successful kidnappings per year is easy to achieve, especially for professional kidnappers

that may be so, but a large Human presence in Hai space has't been a thing for centuries on end; most humans don't know of the Hai, and most that do know show to be the everyday merchant. furthermore, judging by the Pirate Troubles missions
`Human attacks on our systems are very rare, but recently we have experienced an increased number of raids`
 Hai haven't had much issue with Human criminals in the past, so I don't think their people would be getting kidnapped by the thousands. Maybe just like the section I show above, have it be so that very occasionaly some Hai would get kidnapped, but lately the rate of those has been going up considerably?

NomadicVolcano commented:
> Human attacks on our systems are very rare, but recently we have experienced an increased number of raids

The raids of which you speak are pirate raids, in which pirate ships attack a planet or ship openly. Those are understandably rare because: 1) the pirates don't stand a chance, 2) any pirates who are smuggling from Hai space will be pissed off that you may blow their cover, and 3) there are easier targets.

I'm not talking about pirates rushing in with guns blazing. That was my previous attempt  at this atory. I abandoned that because there is no way it would work. Even if it did work, the Hai wouldn't hesitate to send a few Beetles through to crush the pirates if it went on too long.

These are con artists tricking people into entering vulnerable situations and then kidnapping them. That's how most nonconsensual disappearances happen. The only reason they're caught is that they use nanotransceivers so advanced the Hai and humans don't know what to look for.

NomadicVolcano commented:
> maybe not exact numbers or data, but the human you encounter that tells you about the Hai says most of us came here to escape from the chaos and fighting in human space. and that Hai society is so advanced, they need almost no resources to maintain it. , so I'd say it's safe to say the crime there is much rarer than Human space.

Human space isn't like the US in terms of crime rate. It's like traveling through modern-day Syria. Your life is at risk the very second you leave a planet. Kidnapping and pirate raids are frequent everywhere except the innermost systems. Slavery is commonplace. There's a constant fear of Alphas coming back. Secretive corporations that have their own militaries virtually control the government. People feel safer in Syndicate space under an authoritarian corporate regime because at least there they won't be shot in the street. Conditions on Earth are so bad that everyone takes the first opportunity they can to leave. The Free Worlds rebels in part because they're being overrun by pirates and the Republic Parliament doesn't care. 

People are fleeting from an absolute hell. Even if Hai space had ten times the US crime rate, it would seem utopian to humans.

Arachi-Lover commented:
I'm not criticizing the way these guys operate, rather the implication that Hai are kidnapped on the daily, and have been so for decades, and there's not as much as a mention of or an implication for it anywhere. I guess we could work with the large number of kidnappings if there was more a reason given to why it hasn't sparked much uproar or become a common urban legend even among humans there *or* if there was something that let us known this wasn't just something inside this mission chain (thought of maybe some spaceport news, maybe of someone commenting on how they came to visit a Hai friend but they were nowhere to be found, but then those missions would need to go away after the mission line is done I'd guess, so not sure how doable it is)

NomadicVolcano commented:
Spaceport news would work if it was written in the past tense with an unspecified time reference. How does this sound? I can make more of these:

- "My friend went on one of those prison cruises, but his ship was lost in an ion storm. I haven't seen him since."
- "One of my Hai co-workers went to Skillet to arrange a buisness deal. I warned her that part of human space was dangerous. Now I'll probably never see her again."
- "A while ago, a Hai neighbor of mine went to sleep and woke up in a box out in human space. Isn't that the weirdest thing?"

Arachi-Lover commented:
those could work yeah

downRightCoder256 commented:
i don't see why would they have to disappear after the storyline ends. its not like the kidnappings will just stop afterwards

Arachi-Lover commented:
after the Humans and Hai sign the thingy and fleets start guarding both ends, and the customs thing is set up on Allhome, and after the most successful kidnappers were dealt with, I doubt one would hear of a Hai getting kidnapped again for many many decades. Sure there'd still be the randoms that don't follow the Allhome customs protocol, but how many would that be after the Hai know for a fact that such things happened in the first place, and the chore that it was for everyone involved to get it all to stop?

NomadicVolcano commented:
> its not like the kidnappings will just stop afterwards

The network that was arranging the kidnappings has been shut down, and there is a treaty between the Hai, Free Worlds, and Republic to track Hai in human space and provide assistance to them. There's extensive surveillance fleets on both sides of the wormhole checking for Hai or other illegal cargo. Moreover, the kidnappings that did happen nearly sparked a war.

After the treaty, kidnappings across the border will be a rare and politically explosive.

downRightCoder256 commented:
Hm, onto implementing the conditional spaceport news, then.

NomadicVolcano commented:
> Hm, onto implementing the conditional spaceport news, then.

As long as the text is written in the past tense, it will be fine.

downRightCoder256 commented:
I guess, but we'd still need it for other stuff, such as an alt to the #4907, as proposed by Zitchas. I will open an issue nonetheless.

NomadicVolcano commented:
> I guess, but we'd still need it for other stuff, such as an alt to the #4907, as proposed by Zitchas. I will open an issue nonetheless.

See PR #4770 for my solution to that problem.

Arachi-Lover commented on data/human/news.txt:
```diff
@@ -117,15 +141,151 @@ news "human merchant in Hai space"
+			"Hai can terraform planets a hundred times faster than we can. I wonder why they didn't do that at Darkwaste?"
+			"Hai have the most amazing holographic technology. One of them put on a backpack holoprojector. When he turned it on, he looked like an elephant. The legs and tail even tracked his movements! I wonder if they can mimic humans too?"
+			"The Hai have an amazing grasp of quantum chromodynamics. It's why their shield generators are so effective."
+			"The alloys Hai use in their ships is way beyond our technology."
```
`is` -> `are`

NomadicVolcano commented:
I added spaceport news alluding to the pirate and kidnapping problems. I also added references to other stories and aspects of Hai space. Examples:

- **Merchant captain**, "I met some young Hai pilots. It sounds like they start out as poor as we do."
- **Reformed pirate**, "I used to rob people for a living, and I'd be in jail now if I stayed in human space. Here, they've given me a job, and I've been sober for six weeks. This has really turned my life around."
- **Tourist**, "I saw the Hai-Human fusion band Nuclear Martini playing on Luna. Nobody realized the hairy drummer was really a Hai in disguise."
- **Scientist or engineer**, "A while back, my Hai research partner was nearing a breakthrough, and then she vanished. I haven't seen her since. She probably joined the Unfettered."
- **Merchant captain**, "Some Hai were protesting the work week. They want to work only 32 hours a week instead of 40. I used to work 70 hours a week for the Syndicate!"
- **Visiting chef**, "A Hai-Human fusion restaurant near here sells peanut pizza. It sounds gross, but the taste is heavenly."
- **Human citizen**, "A Hai that lives near me went to sleep one night and woke up in a box on Prime. Isn't that weird?"

NomadicVolcano commented:
The checks are failing due to network errors unrelated to my changes.

ravenshining commented:
It looks like you took an image from Wikipedia for your brown dwarf - which is fine, because the image is public domain.  But the image is of a T-type brown dwarf, whereas you've named the file m-brown-dwarf.  I'm working on a PR to add a bunch of star types, and it'd be nice to have things consistent.  I have an M-dwarf and an L-dwarf, but not a T-dwarf, so yours would round things out nicely if you used the proper designation.

NomadicVolcano commented:
> It looks like you took an image from Wikipedia for your brown dwarf - which is fine, because the image is public domain. But the image is of a T-type brown dwarf, whereas you've named the file m-brown-dwarf. I'm working on a PR to add a bunch of star types, and it'd be nice to have things consistent. I have an M-dwarf and an L-dwarf, but not a T-dwarf, so yours would round things out nicely if you used the proper designation.

My neutron star looks more like a protostar. I should probably rename both.

Righthand and I will need a lot of dim star types for the Whispering Void since it is in deep space where the stars are not visible from a long distance away. Also, we'll need several pulsars for storyline reasons. He and I are putting together a repo for the SMC expansion, so we have a common area to assemble things for PRs. If you think you could help with the star types, then please contact him about that. It'll let me focus on critters, ships, stories, and outfits.

https://github.com/endless-sky-smc

NomadicVolcano commented:
> It looks like you took an image from Wikipedia for your brown dwarf - which is fine, because the image is public domain. But the image is of a T-type brown dwarf, whereas you've named the file m-brown-dwarf. I'm working on a PR to add a bunch of star types, and it'd be nice to have things consistent. I have an M-dwarf and an L-dwarf, but not a T-dwarf, so yours would round things out nicely if you used the proper designation.

Do you have a planned naming convention? If so, please let me know what to call the protostar and T-dwarf in `images/stars/*png`

ravenshining commented:
Sure.  Here's the current state of the images/star folder on my branch:
![stars](https://user-images.githubusercontent.com/31712538/77663965-bf299a00-6f21-11ea-95d8-bbba9c35a544.png)

NomadicVolcano commented:
> Sure. Here's the current state of the images/star folder on my branch:

The dwarves are of a very different style, and with less detail, than the one I stole from NASA.

Also, here's the original source:

https://exoplanets.nasa.gov/resources/114/forecast-for-exotic-weather/

I'll go with "protostar-green-purple" for the protostar, unless you have a better idea.

ravenshining commented:
Yeah, those dwarves are to-scale with the other stars - on a logarithmic scale, diameter = 120 * (multiples of Sol diameter)^(1/e) - so the sprites are small, about 22 pixels across for the tiniest.  At the moment they're just resizes and rehues of ES's stock stars, as my focus is on map.txt at the moment, but I did make a planet-sized L-dwarf:
![](https://github.com/ravenshining/endless-sky/blob/proper-stars/images/planet/l-browndwarf.png?raw=true)

NomadicVolcano commented:
Having them visible on the screen is more important than having them be to scale. If the star is so small you can't see it, then there's no point in making the art in the first place.

NomadicVolcano commented:
Please try installing the PR and go visit the three new systems. Look at the sizes I chose. I've made the dwarves large enough so you can see their detail but small enough that they're tiny on the screen. As long as we don't have them in the same system as a larger star, the scale shouldn't be a problem.

ravenshining commented:
The point of what I was doing was precisely so you could put a dwarf in the same system as a larger star, which is part of why I used a logarithmic scale instead of a linear scale.  It was in recognition of the lack of ability to show detail that I started making planet-sized versions.  You could make larger-scale dwarves as planets - a brown dwarf is only the size of Jupiter after all - or have a separate name, like "t-dwarf-enlarged.png"

Here's an extraneous m-dwarf added to an Alpha Centauri with star types corrected to show what I'm going for:
![quadstar](https://user-images.githubusercontent.com/31712538/77666459-12e9b280-6f25-11ea-92f2-44fd3d28f1e0.png)

NomadicVolcano commented:
That is a fine plan. Do you want to make a T-dwarf and give it to me? Then the art style would be consistent. I can credit you in the commit.

ravenshining commented:
Sure, I'm working on some full-res dwarves now.  Here's a WIP m-dwarf, self-lit version for use as a lone dwarf.  Feedback welcome while it's still in layers:
![dwarfprogress](https://user-images.githubusercontent.com/31712538/77702001-97efbe80-6f5b-11ea-8e5b-9d7315e45038.png)

NomadicVolcano commented:
> Sure, I'm working on some full-res dwarves now. Here's a WIP m-dwarf, self-lit version for use as a lone dwarf. Feedback welcome while it's still in layers:

It is excellent. Do you have any comments on my protostar?

ravenshining commented:
It is beautiful, and yes, definitely better as a protostar than a neutron star.  However, while the purple and green is really cool, it's probably a bit too exotic.

Here's an in-game shot after getting some feedback on discord.  Unless others have more feedback I'll move on to dimmer dwarves
![m-dwarftest](https://user-images.githubusercontent.com/31712538/77711069-02145d80-6f74-11ea-84fc-7ad412b95c2e.png)

NomadicVolcano commented:
I've made a few changes. The  notable ones:

1. No dwarves or neturon stars; I'm using two proto-stars. I've cleaned them up a bit.

2. Many tiny changes to allow MasterOfGrey's story to handle the humans "discovering" the Hai

3. Workaround for a game bug. Your Unfettered Hai escorts blithely ignore the tribute defense fleets that are attacking you. I've replaced the fleets with 20 NPC ships and 10 tribute defense ships. The player is warned they'll have to fight the tribute defense ships. The alternative is to have only NPC ships, but that makes the battle nigh-impossible since all 30 ships jump-camp you. Lowering it to 20 is enough to use the 15 beetles as a shield for the first few seconds. This is a reasonable compromise.

NomadicVolcano commented:
I'm still pondering better ways to handle the final battle. I have to strike a balance between CPU load, instant death by jump-camping, the capabilities of fifteen shield beetles, the expected abilities of a player with 3000 combat rating, and the limitations of pirate capital ships. 

There's another route I can take. We know the pirates captured three Korath Raiders. I could use those to defend Paradise, but it would raise questions about why the pirates weren't using those to raid planets instead.

NomadicVolcano commented:
I'm going to merge a bunch of changes MasterOfGrey and I have been working on from the Hai Reveal work over to this branch in the next few days. I'm temporarily changing this back to a Draft PR to make sure it isn't reviewed until then.

NomadicVolcano commented:
I don't feel like dealing with this PR any more, so I'm closing it. The changes are already in #4952.

Zitchas commented:
By "the changes are already in 4952," do you mean all the missions, hai stuff (news, variants, etc), and the smc stuff, are all in 4952?

NomadicVolcano commented:
> By "the changes are already in 4952," do you mean all the missions, hai stuff (news, variants, etc), and the smc stuff, are all in 4952?

It has been substantially rearranged and incorporated into what is best called the Hai Main Story. With the changes in #4952, the Hai are no longer a placeholder side with no story or personality. They've grown up and have substance to them. This story is about half of that.

The SMC is not in this PR; it will be split into multiple PRs in the future. The first will be the rest of the Whispering Void, but I need the Philosopher's Paradise story before I can write any stories in the Whispering Void. Otherwise the Whispering Void is just some critters flying around in space.

Zitchas commented:
OK. So long as this material isn't being abandoned - just a change in methodology - then I'm good with this being closed. I just don't want to lose this stuff.
